### PR TITLE
feat(doc integrations): Add target_id and target_type columns

### PR DESCRIPTION
**Pull request by ceorourke**
This is the first of a series of migrations to update the `IntegrationFeature` table to be useable by both sentry apps and doc integrations.

- [x] Add `target_id` and `target_type` columns, make `sentry_app` column nullable (this PR)

- [x] Dual write  to both columns (https://github.com/getsentry/sentry/pull/30548)

- [x] Backfill all existing rows into the new columns (https://github.com/getsentry/sentry/pull/30563)

- [x] Update any existing code to not reference the `sentry_app` column (https://github.com/getsentry/sentry/pull/30569)

- [x] Remove the `sentry_app` column from the table, but mark the state only as removed in the migration (https://github.com/getsentry/sentry/pull/30572)

- [x] Remove the `sentry_app` column (https://github.com/getsentry/sentry/pull/30579)

- [x] Make `target_id` not null (https://github.com/getsentry/sentry/pull/30583)

### CODE REVIEW

File: src/sentry/migrations/0257_add_target_id_and_type_to_integrationfeature.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    operations = [
+        migrations.AddField(
+            model_name="integrationfeature",
+            name="target_id",
+            field=sentry.db.models.fields.bounded.BoundedBigIntegerField(null=True),
+        ),
+        migrations.AddField(
+            model_name="integrationfeature",
+            name="target_type",
+            field=sentry.db.models.fields.bounded.BoundedPositiveIntegerField(default=0),
```
Comment (wedamija) [inline src/sentry/migrations/0257_add_target_id_and_type_to_integrationfeature.py:44]: This table is tiny right? If so this should be safe.

File: src/sentry/models/integrationfeature.py [hunk 2]
```diff
@@ -68,10 +75,24 @@ def description(cls, feature, name):
...
+
 class IntegrationFeature(Model):
     __include_in_export__ = False
 
-    sentry_app = FlexibleForeignKey("sentry.SentryApp")
+    sentry_app = FlexibleForeignKey("sentry.SentryApp", null=True)
+    # the id of the sentry_app or doc_integration
+    # TODO(CEO): change this to null=False
+    target_id = BoundedBigIntegerField(null=True)
+    target_type = BoundedPositiveIntegerField(
```
Comment (wedamija) [inline src/sentry/models/integrationfeature.py:90]: This needs a default, or to be nullable.

**Review by wedamija — COMMENTED** [review]
We need one extra step in the migration plan: Before `Backfill all existing rows into the new columns` we need to start dual writing to both columns

File: src/sentry/migrations/0257_add_target_id_and_type_to_integrationfeature.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/migrations/0257_add_target_id_and_type_to_integrationfeature.py:44]: It's 20,689 rows, maybe not *tiny* but ... small?

Comment (wedamija) [reply src/sentry/migrations/0257_add_target_id_and_type_to_integrationfeature.py:44]: Yeah, that's small enough.

**Review by wedamija — APPROVED** [review]
Since we're adding the default and backfilling via Django, there's a point in time where the migration will have run, but old code is still out. This means that while the deploy is in progress, creating any new `IntegrationFeature` rows will fail, but reading them will work fine. I don't know when we create these, but up to you if that's important - if it is, then you can make `target_type` null to avoid any issues.

**Pull request merged**